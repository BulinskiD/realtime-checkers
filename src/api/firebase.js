import app from "firebase/app";
import "firebase/storage";
import "firebase/auth";
import "firebase/firestore";

var firebaseConfig = {
  apiKey: "AIzaSyD3uWma_iug5o_0w6EmchIXp4qd054HqcE",
  authDomain: "realtime-checkers.firebaseapp.com",
  databaseURL: "https://realtime-checkers.firebaseio.com",
  projectId: "realtime-checkers",
  storageBucket: "realtime-checkers.appspot.com",
  messagingSenderId: "100295366319",
  appId: "1:100295366319:web:ec1c182d83b660e8"
};

// Initialize Firebase
app.initializeApp(firebaseConfig);

export const auth = app.auth();
export const firestore = app.firestore();
