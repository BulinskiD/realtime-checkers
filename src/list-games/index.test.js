import React from 'react';
import {firestore} from "../api/firebase";
import {StaticRouter} from 'react-router-dom';
import ListGames from './index';
import {mount, shallow} from 'enzyme';

jest.mock('../api/firebase');

firestore.collection = jest.fn();

describe("ListGames", () => {
   it('Should match snapshot for given props', () => {
       const dataResp = jest.fn();
       dataResp.mockReturnValue({title: 'test'});
       const response = {docs: [{id: '11', data: dataResp}]};
       const onSnapshot = jest.fn(func => func(response));
       firestore.collection.mockReturnValue({onSnapshot});
       const component = mount(<StaticRouter><ListGames /></StaticRouter>);

       expect(firestore.collection).toHaveBeenCalledTimes(1);
       expect(onSnapshot).toHaveBeenCalledTimes(1);
       expect(dataResp).toHaveBeenCalledTimes(1);
       expect(component).toMatchSnapshot();
   });

    it('Should display title in list and correct path', () => {
        const dataResp = jest.fn();
        dataResp.mockReturnValue({title: 'test'});
        const response = {docs: [{id: '11', data: dataResp}]};
        const onSnapshot = jest.fn(func => func(response));
        firestore.collection.mockReturnValue({onSnapshot});
        const component = mount(<StaticRouter><ListGames /></StaticRouter>);

        expect(component.find('a').props().href).toBe('/game/11');
        expect(component.find('div.list-group-item').text()).toBe('test');
    });
});
