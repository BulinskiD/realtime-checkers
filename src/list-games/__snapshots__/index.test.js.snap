// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListGames Should match snapshot for given props 1`] = `
Array [
  <h4
    className="my-3"
  >
    DostÄ™pne gry: 
  </h4>,
  <hr />,
  <ForwardRef
    className="d-flex justify-content-center"
  >
    <Bootstrap(ListGroup)
      className="d-flex justify-content-center"
      onSelect={[Function]}
    >
      <ContextTransform(AbstractNav)
        as="div"
        className="d-flex justify-content-center list-group"
        onSelect={[Function]}
      >
        <div
          className="d-flex justify-content-center list-group"
          onKeyDown={[Function]}
        >
          <a
            href="/game/11"
            onClick={[Function]}
          >
            <Bootstrap(ListGroupItem)
              className="text-center"
            >
              <ForwardRef
                as="div"
                className="text-center list-group-item"
                disabled={false}
                eventKey={null}
                onClick={[Function]}
              >
                <div
                  className="text-center list-group-item"
                  data-rb-event-key={null}
                  disabled={false}
                  onClick={[Function]}
                >
                  test
                </div>
              </ForwardRef>
            </Bootstrap(ListGroupItem)>
          </a>
        </div>
      </ContextTransform(AbstractNav)>
    </Bootstrap(ListGroup)>
  </ForwardRef>,
]
`;
